rotkehlchen/chain/evm/decoding/omnibridge/decoder.py

```diff
@@ -0,0 +1,130 @@
+            userdb=self.evm_inquirer.database,
+            evm_address=hex_or_bytes_to_address(context.tx_log.topics[1]),
+            chain_id=self.evm_inquirer.chain_id,
+            evm_inquirer=self.evm_inquirer,
+        )
+        amount = asset_normalized_value(
+            amount=hex_or_bytes_to_int(context.tx_log.data[0:32]),
+            asset=bridged_asset,
+        )
+        expected_event_type, new_event_type, from_chain, to_chain, expected_location_label = bridge_prepare_data(  # noqa: E501
+            tx_log=context.tx_log,
+            deposit_topics=(TOKENS_BRIDGING_INITIATED,),
+            source_chain=self.source_chain,
+            target_chain=self.target_chain,
+            from_address=from_address,
+            to_address=to_address,
+        )
+
+        for event in context.decoded_events:
+            if bridged_asset == A_WETH and event.asset == A_ETH and event.balance.amount == amount:
+                if event.event_type == HistoryEventType.RECEIVE:
+                    # Use transaction.from_address in ETH withdrawals since tx_log.topics[2] has other data  # noqa: E501
+                    expected_location_label = context.transaction.from_address
+                    from_address = expected_location_label
+                    to_address = expected_location_label
+                expected_address = hex_or_bytes_to_address(context.tx_log.topics[2])
+                expected_asset = A_ETH
+            else:
+                expected_address = self.bridge_address
+                expected_asset = bridged_asset
+
+            if (
+                event.event_type == expected_event_type and
+                event.location_label == expected_location_label and
+                event.address in (expected_address, ZERO_ADDRESS) and
+                event.asset == expected_asset and
+                event.balance.amount == amount
+            ):
+                bridge_match_transfer(
+                    event=event,
+                    from_address=from_address,
+                    to_address=to_address,
+                    from_chain=from_chain,
+                    to_chain=to_chain,
+                    amount=event.balance.amount,
+                    asset=bridged_asset,
+                    expected_event_type=expected_event_type,
+                    new_event_type=new_event_type,
+                    counterparty=GNOSIS_CPT_DETAILS,
+                )
+                break
+        else:
+            log.error(f'Could not find the transfer event for bridging to {to_address} in {context.transaction}')  # noqa: E501
+
+        return DEFAULT_DECODING_OUTPUT
+
+    # -- DecoderInterface methods
+
+    def addresses_to_decoders(self) -> dict[ChecksumEvmAddress, tuple[Any, ...]]:
+        return {self.bridge_address: (self._decode_bridge_tokens,)}
+
+    @staticmethod
+    def counterparties() -> tuple[CounterpartyDetails, ...]:
+        return (GNOSIS_CPT_DETAILS,)
```

Comment:
I am not sure if this should be the counterparty or if we should add a new "omnibridge" counterparty.

Perhaps keep it as gnosis? As omnibridge used to have its own logo and ui: 

![2024-08-17_16-05](https://github.com/user-attachments/assets/be42e29d-40f9-422a-a43f-2e40d0fac2c6)
https://omni.legacy.gnosischain.com/bridge

But now it's become the only official gnosis bridge from ethereum: https://bridge.gnosischain.com/
![2024-08-17_16-06](https://github.com/user-attachments/assets/67aab798-18bc-428f-af1d-8c6a4519db1d)